diff --git a/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp b/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp
index c77db8ed2..f11b5b308 100644
--- a/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp
+++ b/mm/2s2h/Rando/ActorBehavior/ActorBehavior.cpp
@@ -7,7 +7,7 @@ extern "C" {
 }
 
 // This is kind of a catch-all for things that are simple enough to not need their own file.
-void MiscVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, void* optionalArg) {
+int MiscVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, va_list optionalArg) {
     switch (id) {
         case VB_MADAME_AROMA_ASK_FOR_HELP:
             *should = !CHECK_WEEKEVENTREG(WEEKEVENTREG_BOMBERS_NOTEBOOK_EVENT_RECEIVED_KAFEIS_MASK);
@@ -26,6 +26,7 @@ void MiscVanillaBehaviorHandler(GIVanillaBehavior id, bool* should, void* option
             *should = false;
             break;
     }
+    return 0;
 }
 
 ActorExtensionId randoCheckIdActorExt = 0;
